src/store/reducers/settings/settings.ts

```diff
@@ -115,13 +120,14 @@ export const setSettingValue = (
     name: string,
     value: string,
 ): ThunkAction<void, RootState, unknown, SetSettingValueAction> => {
-    return (dispatch, getState) => {
+    return (dispatch) => {
         dispatch({type: SET_SETTING_VALUE, data: {name, value}});
-        const {singleClusterMode} = getState();
-        if (singleClusterMode) {
-            localStorage.setItem(name, value);
-        } else {
+
+        // If there is no settingsApi, use localStorage
+        if (settingsApi) {
             window.api.postSetting(name, value);
+        } else {
+            localStorage.setItem(name, value);
```

Comment:
Potentially this operation may fail, sp lets wrap it with `try ... catch`.

Reply:
What should be in `catch` section in this case?